core/parsing_handler.py

```diff
@@ -53,28 +51,12 @@ def parse_and_validate(directory):
     return sequencing_run
 
 
-def run_is_new(directory):
+def get_run_status(directory):
     """
-    Given a run directory, returns True or False if run is new
+    Given a run directory, returns a DirectoryStatus object created by the parser
     :param directory:
-    :return:
+    :return: DirectoryStatus
     """
     parser_instance = get_parser_from_config()
     status = parser_instance.find_single_run(directory)
-    return _run_is_new_via_directory_status(status)
-
-
-def _run_is_new_via_directory_status(directory_status):
-    """
-    Checks if a run directory is new or not
-
-    :param directory_status: a DirectoryStatus object created by the progress module
-    :return: True or False
-    """
-    parser_instance = get_parser_from_config()
-    result = upload_status.get_directory_status(
-        directory_status.directory,
-        parser_instance.get_sample_sheet_file_name()
-    )
```

Comment:
This function was bad to begin with, we are doing steps that only the lower parser level should be doing.